source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.cc

```diff
@@ -67,33 +81,44 @@ Http::FilterTrailersStatus Http1BridgeFilter::encodeTrailers(Http::ResponseTrail
   }
 
   if (do_bridging_) {
-    // Here we check for grpc-status. If it's not zero, we change the response code. We assume
-    // that if a reset comes in and we disconnect the HTTP/1.1 client it will raise some type
-    // of exception/error that the response was not complete.
-    const Http::HeaderEntry* grpc_status_header = trailers.GrpcStatus();
-    if (grpc_status_header) {
-      uint64_t grpc_status_code;
-      if (!absl::SimpleAtoi(grpc_status_header->value().getStringView(), &grpc_status_code) ||
-          grpc_status_code != 0) {
-        response_headers_->setStatus(enumToInt(Http::Code::ServiceUnavailable));
-      }
+    // We're bridging, so we need to process trailers and set the grpc status and message from
+    // the those input trailers.
+    doTrailers(trailers, /*set_grpc_status_and_message=*/true);
+  }
+
+  // NOTE: We will still write the trailers, but the HTTP/1.1 codec will just eat them and end
+  //       the chunk encoded response which is what we want.
+  return Http::FilterTrailersStatus::Continue;
+}
+
+void Http1BridgeFilter::doTrailers(const Http::ResponseHeaderOrTrailerMap& trailers,
```

Comment:
Should this just be split into two functions instead of passing a book param?

Reply:
I'll split this into doTrailers and doTrailersOnlyResponse

Reply:
On second thought, I think splitting this out will just mean there's a second path and a small amount of duplication, which doesn't feel strictly better than the current solution, even if a bit over-parameterized - WDYT?

Reply:
I went ahead and made the split - let me know if you think it is better!

Reply:
My thinking was to split the functions up in terms of what they were doing, not in terms of when they'd be called. So something like 

```
doResponseTrailers();
updateGrpcStatusAndMessage();
```

Or is there a good reason to have to do it in one function? afaik a lookup of any of the headers should be O(1)

Reply:
No good reason to do it in one function other than a weak attempt at reducing code duplication.

re: function split up and names, let me look into that

Reply:
I reorganized it a bit - take a look